hcl2template/types.build.go

```diff
@@ -125,3 +132,356 @@ func (p *Parser) decodeBuildConfig(block *hcl.Block) (*BuildBlock, hcl.Diagnosti
 
 	return build, diags
 }
+
+// A CoreHCL2Build struct represents a single build job, the result of which should
+// be a single machine image artifact. This artifact may be comprised of
+// multiple files, of course, but it should be for only a single provider (such
+// as VirtualBox, EC2, etc.).
+type CoreHCL2Build struct {
```

Comment:
I didn't want to change anything in the original CoreBuild so I copied and pasted everything and made the necessary changes.

Reply:
The only differences are the code for decoding and preparing provisioners and post-processors

Reply:
Would it be a good idea to embed a CoreBuild to lower the number of necessary copy pastas ? For example :

```go
type CoreHCL2Build struct {
  packer.CoreBuild
}
```

Reply:
I did this at first but then some of the attributes in CoreBuild were private and I had to change them everywhere. It was more to make the diff smaller with only the interested files. I can do it later in another PR after this is working perfectly.

Reply:
Got it, I think the other idea of the embedded func that does this is a bit better, I'll let you decide :D

Reply:
Yes! The other idea was much better! I've implemented already and it looks really good 😄